[Docs] improve the LoRA doc. (#4838)
huggingface/diffusers: docs/source/en/training/lora.md

* LoRA parameters that have separate identifiers for the UNet and the text encoder such as: [`"sayakpaul/dreambooth"`](https://huggingface.co/sayakpaul/dreambooth).

> **Note** that it is possible to provide a local directory path to [`~diffusers.loaders.LoraLoaderMixin.load_lora_weights`] as well as [`~diffusers.loaders.UNet2DConditionLoadersMixin.load_attn_procs`]. To know about the supported inputs, refer to the respective docstrings.

Reviewer: For things we want to note/draw users attention to, may be better to use our nice callouts :)

```suggestion
<Tip>

You can also provide a local directory path to [`~diffusers.loaders.LoraLoaderMixin.load_lora_weights`] as well as [`~diffusers.loaders.UNet2DConditionLoadersMixin.load_attn_procs`].

</Tip>
```

## Stable Diffusion XL

We support fine-tuning with [Stable Diffusion XL](https://huggingface.co/papers/2307.01952). Please refer to the following docs:

* [text_to_image/README_sdxl.md](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/README_sdxl.md)
* [dreambooth/README_sdxl.md](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/README_sdxl.md)

## Unloading LoRA parameters

You can call [`~diffusers.loaders.LoraLoaderMixin.unload_lora_weights`] on a pipeline to unload the LoRA parameters.

## Fusing LoRA parameters

You can call [`~diffusers.loaders.LoraLoaderMixin.fuse_lora`] on a pipeline to merge the LoRA parameters with the original parameters of the underlying model(s). This can lead to a potential speedup in the inference latency.

## Unfusing LoRA parameters

> You can call [`~diffusers.loaders.LoraLoaderMixin.unfuse_lora`] on a pipeline to reverse the effects of `fuse_lora()`.

Reviewer: Since this is so short and not that distinct of a topic from "Fusing LoRA parameters", it would be better to merge the two sections together.

```suggestion
To undo `fuse_lora`, call [`~diffusers.loaders.LoraLoaderMixin.unfuse_lora`] on a pipeline.
```

## Supporting different LoRA checkpoints from Diffusers

> LoRA trainers such as [Kohya](https://github.com/kohya-ss/sd-scripts/) and [TheLastBen](https://github.com/TheLastBen/fast-stable-diffusion) are quite popular in the community. We try to provide a seamless ability to load those checkpoints in Diffusers. In this section, we outline the details and the limitations of the current API. 

Reviewer: ```suggestion
🤗 Diffusers supports loading checkpoints from popular LoRA trainers such as [Kohya](https://github.com/kohya-ss/sd-scripts/) and [TheLastBen](https://github.com/TheLastBen/fast-stable-diffusion). In this section, we outline the current API's details and limitations. 
```

Thanks to [@isidentical](https://github.com/isidentical) for helping us on integrating this feature.

<Tip warning={true}>

**Known limitations specific to the Kohya LoRAs**: 

* When images don't looks similar to other UIs, such as ComfyUI, it can be because of multiple reasons, as explained [here](https://github.com/huggingface/diffusers/pull/4287/#issuecomment-1655110736).
* We don't fully support [LyCORIS checkpoints](https://github.com/KohakuBlueleaf/LyCORIS). To the best of our knowledge, our current `load_lora_weights()` should support LyCORIS checkpoints that have LoRA and LoCon modules but not the other ones, such as Hada, LoKR, etc. 

</Tip>

### TheLastBen

> Here is an example:

Reviewer: Maybe also add some introductory text here instead of abruptly dropping the user straight into the code. Something like:

```md
You can also load checkpoints from TheLastBen by specifying the model id and filename in the [`~diffusers.loaders.LoraLoaderMixin.load_lora_weights`] function:
